File: daily/Linked List/143-Reorder-list.py
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        """
        Do not return anything, modify head in-place instead.
        """
        nodes = []
        curr = head
        while curr:
            nodes.append(curr)
            curr = curr.next
        
        left, right = 0, len(nodes) - 1
        while left < right:
            nodes[left].next = nodes[right]
            left += 1

            if left == right:
                break
            
            nodes[right].next = nodes[left]
            right -= 1
        nodes[left].next = None

File: daily/Linked List/143-Reorder-list.go
package daily

func reorderList(head *ListNode)  {
    nodes := []*ListNode{}
    curr := head
    for curr != nil {
        nodes = append(nodes, curr)
        curr = curr.Next
    }

    left, right := 0, len(nodes) - 1
    for left < right {
        nodes[left].Next = nodes[right]
        left++

        if left == right {
            break
        }

        nodes[right].Next = nodes[left]
        right--
    }
    nodes[left].Next = nil
}
